website/docs/cdktf/create-and-deploy/deployment-patterns.mdx

```diff
@@ -0,0 +1,169 @@
+
+You can provision your infrastructure using either CDKTF CLI commands (e.g., `cdktf deploy`) or by synthesizing your application into a Terraform configuration file and using Terraform directly. Both approaches allow you to use Terraform Cloud and integrate with a Continous Integration (CI) pipeline.
+
+### Run CDKTF CLI Commands
+
+We recommend using [CDKTF CLI commands](/cdktf/cli-reference/commands) in the following use cases:
+
+- **Your application performs additional tasks** For example, your application may build a Docker image or start a build pipeline locally through synchronous APIs. You might also be using the null provider to execute these actions when a resource is created (e.g., after the Docker registry has been created). Running `cdktf deploy` triggers all of these actions at runtime in addition to using Terraform to provision your infrastructure.
+
+- **Your application contains stacks with complex dependencies.** Each [stack](/cdktf/concepts/stacks) in a CDKTF application is like a separate Terraform working directory. Terraform respects complex dependencies within the same configuration, but it is not designed to manage complex dependencies between multiple working directories. For example, if you define a stack A that depends on data from stack B and stack C, Terraform runs will fail. This is because Terraform will not understand that it must wait to build stack A until both stack B and stack C are completed.
+
+### Run `cdktf synth` and Use Terraform Directly
+
+We recommend synthesizing your CDKTF application and using Terraform directly when you need to integrate into existing Terraform workflows. Your organization may already use Terraform Cloud or have integrated Terraform into CI pipelines. In these cases, you may want to run `cdktf synth` and set the output folder to a particular directory that you can reference in your existing workflows.
+
+## Using Terraform Cloud
+
+Terraform Cloud lets you review the current state of your infrastructure and run progress in the Terraform Cloud UI. You can also use Terraform Cloud as the source of secrets via [workspace variables](/cloud-docs/workspaces/variables). If you use the local execution mode or any other backend you must specify the credentials for your providers locally through environment variables.
+
+To use Terraform Cloud with a [CLI-driven Run Workflow](/cloud-docs/run/cli), you must enable [remote execution](/cloud-docs/run#remote-operations) in your Terraform Cloud workspace.
+
+~> **Warning:** Remote execution with non-interactive workflows requires auto-approved deployments. Minimize risk of unpredictable infrastructure changes and configuration drift by making sure that no one can change your infrastructure outside of your automated build pipeline. Refer to [Non-Interactive Workflows](/cloud-docs/run/cli#non-interactive-workflows) for details.
+
+To run Terraform Cloud with a [VCS-Driven Run Workflow](/cloud-docs/run/ui) you must check in your synthesized code.
```

Comment:
Do we need to say this bit about VCS? i'm wondering if that is implied?

Reply:
What do you mean? Like what VCS is?

Reply:
oh no i was just wondering if we needed to tell folks that they have to check their synthesized code in if they're using VCS. If you want to keep this in I'm okay with that, but I was also wondering if we NEED to tell people that or if folks will already know that you have to check things in to use VCS. Sorry, I was not clear there at all!

Reply:
I'd rather error on the side of being overly explicit here

Reply:
okay I support that then :)